sys/arm64/arm64/exec_machdep.c

```diff
@@ -762,7 +762,11 @@ set_mcontext(struct thread *td, mcontext_t *mcp)
 
 	tf->tf_sp = mcp->mc_capregs.cap_sp;
 	tf->tf_lr = mcp->mc_capregs.cap_lr;
-	tf->tf_elr = mcp->mc_capregs.cap_elr;
+	if (SV_PROC_FLAG(td->td_proc, SV_UNBOUND_PCC))
```

Comment:
Would it be worth pushing the handling for `SV_UNBOUND_PCC` into `trapframe_set_elr`?

Reply:
That would do awful things for nested faults in a purecap kernel, no?

Reply:
Oof, yes.  One could do gross things checking the VA for user vs kernel, but probably best not to.